optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -111,8 +124,118 @@ def _from_pretrained(
+                raise AssertionError("Please use neural_compressor version > 3.0.")
+            if is_ipex_version("<", "2.3.1") and use_xpu:
+                raise AssertionError("Please use intel_extension_for_pytorch version >= 2.3.1.")
+
+            if use_xpu:
+                # TODO: if low_cpu_mem_uasge is True, gptj will have accuracy issue on CPU device.
+                kwargs["low_cpu_mem_usage"] = True
+                kwargs["device_map"] = "cpu"
+                try:
+                    model = cls.auto_model_class.from_pretrained(
+                        model_id,
+                        *model_args,
+                        config=config,
+                        **kwargs,
+                    )
+                    model.config.update({"low_cpu_mem_usage": True})
+                except NotImplementedError:
+                    logger.info(
+                        "Failed to load models with `low_cpu_mem_usage` specified, "
+                        "will fall to traditional load method with higher memory consumption."
+                    )
+                    kwargs["low_cpu_mem_usage"] = False
+                    model = cls.auto_model_class.from_pretrained(
+                        model_id,
+                        *model_args,
+                        config=config,
+                        **kwargs,
+                    )
+                    model.config.update({"low_cpu_mem_usage": False})
+                    quantization_config.post_init_xpu()
+            else:
+                kwargs["low_cpu_mem_usage"] = True
+                model = cls.auto_model_class.from_pretrained(
+                    model_id,
+                    *model_args,
+                    config=config,
+                    **kwargs,
+                )
+                model.config.update({"low_cpu_mem_usage": True})
+                quantization_config.post_init_cpu()
+            model.eval()
+
+            if use_xpu:
+                assert hasattr(torch, "xpu") and torch.xpu.is_available(), "There is no xpu device in this system!"
+                quantization_config.update(**{"device": "xpu"})
+                quantization_config.post_init_xpu()
+            if (
+                not torch.cuda.is_available() or device_map == "cpu" or device_map == torch.device("cpu")
+            ) and model.config.model_type == "chatglm":
+                model = model.float()
+            model = convert_to_quantized_model(model, quantization_config, device=device_map)
+            quantization_config.remove_redundant_parameters()
+            model.config.quantization_config = quantization_config
+            # add quantization_config and save_low_bit to pretrained model dynamically
+            model.device_map = device_map
+            model.quantization_config = quantization_config
+            model.save_pretrained = types.MethodType(save_low_bit, model)
+            logger.info("WeightOnlyQuant done.")
+            return model
```

Comment:
```suggestion
            return cls(
                        model, config=config, model_save_dir=model_save_dir, **kwargs
                    )
```

Reply:
I don't want to return INCModel, I would like to return PretrainedModel.